iris/iris.go

```diff
@@ -140,3 +150,34 @@ func (c *Client) NewRequestWithContext(ctx context.Context, r *http.Request, met
 
 	return req, nil
 }
+
+func GetErr(res *http.Response) error {
+	if res.StatusCode < 400 {
+		return nil
+	}
+
+	var err siber.JSendError
+	if err := json.NewDecoder(res.Body).Decode(&err); err != nil {
+		panic(err)
+	}
+	return err
+}
+
+func GetResponse(res *http.Response, v interface{}) (interface{}, error) {
+	var buffer bytes.Buffer
+	bodyReader := io.TeeReader(res.Body, &buffer)
+
+	var j jSendSuccess
+	err := json.NewDecoder(bodyReader).Decode(&j)
+	if err != nil {
+		return err, nil
+	}
+
+	reflect.TypeOf(v)
+	err = mapstructure.Decode(j.Data, &v)
```

Comment:
this will most probably not set anything. 
1. It has issue with strings and numbers
2. it might have an issue with `Data`  vs `data`(which is what you're targetting)

Please check the `mapstructure` API for how to solve these issues it

Reply:
Based on the test, you're passing a reference to a value, won't passing another reference to that reference lead to unexpected results?